{"info": {"author": "Rodrigo Cacilh\u03b1\u03c2", "author_email": "batalema@cacilhas.info", "bugtrack_url": null, "classifiers": ["Environment :: Web Environment", "License :: OSI Approved :: BSD License", "Operating System :: OS Independent", "Programming Language :: Python", "Programming Language :: Python :: 2.7", "Programming Language :: Python :: 3", "Programming Language :: Python :: Implementation :: PyPy", "Topic :: Software Development :: Libraries :: Python Modules"], "description": "=====================\n        Flask-ReportableError\n        =====================\n        \n        Introduction\n        ------------\n        \n        Flask-ReportableError is a Flask extension for handling errors that can\n        be reported to the web client.\n        \n        \n        Documentation\n        -------------\n        \n        In order to use Flask-ReportableError, you must include the following\n        statement in you application start script::\n        \n            from flask import Flask\n            import flask_reportable_error\n        \n            app = Flask(__name__)\n            app.config.from_envvar('FLASK_SETTINGS', silent=True)\n            flask_reportable_error.init(app)\n        \n        \n        API\n        ---\n        \n        - ``flask_reportable_error.init()``:\n          context initializer, it must receive the application as parameter.\n        \n        \n        - ``flask_reportable_error.mixin``:\n          class decorator that declares the decorated class as mixin with\n          reportable exceptions.\n        \n        \n        - ``flask_reportable_error.reportable()``:\n          factory to create reportable exception classes. For example::\n        \n            raise reportable(ValueError)('invalid data received')\n        \n        \n        - ``flask_reportable_error.ReportableErrorMixin``:\n          mixin for reportable exception classes.\n        \n          - ``report()``:\n            method that returns the reportable string \u2013\u00a0can be overridden.\n        \n          - ``status_code``:\n            property representing the numeric status code \u2013\u00a0can be set at\n            instance level.\n        \n          - ``template``:\n            attribute refering to custom template name.\n        \n          - ``headers``:\n            attribute refering to custom headers.\n        \n          - ``type_name``:\n            attribute set by ``reportable()`` on inheritance, with the name of\n            the original exception class.\n        \n        \n        Settings\n        --------\n        \n        The Flask settings may contain the key ``REPORTABLE_ERROR``, that\u2019s a\n        dictionary with the following keys:\n        \n        - ``LOGLEVEL``:\n          the logging level. If not supplied, Flask-ReportableError uses\n          ``logging.ERROR``.\n        \n        - ``DEFAULT_STATUS_CODE``:\n          the default numeric status code for reportable exception classes. By\n          default it\u2019s 500.\n        \n        - ``TEMPLATE``:\n          the name of a template to be used instead of none. The context of the\n          template will contain the exception as value of the ``exc`` key.\n        \n        - ``HEADERS``:\n          a dictionary to be used as default headers instead of ``{}``.", "description_content_type": null, "docs_url": null, "download_url": "https://github.com/Montegasppa/Flask-ReportableError/archive/next_release.zip", "downloads": {"last_day": -1, "last_month": -1, "last_week": -1}, "home_page": "https://github.com/Montegasppa/Flask-ReportableError", "keywords": null, "license": "BSD", "maintainer": null, "maintainer_email": null, "name": "Flask-ReportableError", "package_url": "https://pypi.org/project/Flask-ReportableError/", "platform": "any", "project_url": "https://pypi.org/project/Flask-ReportableError/", "project_urls": {"Download": "https://github.com/Montegasppa/Flask-ReportableError/archive/next_release.zip", "Homepage": "https://github.com/Montegasppa/Flask-ReportableError"}, "release_url": "https://pypi.org/project/Flask-ReportableError/0.4.3/", "requires_dist": null, "requires_python": null, "summary": "handle errors that can be reported to the web client", "version": "0.4.3", "yanked": false, "html_description": "<div class=\"project-description\">\n            =====================<br>        Flask-ReportableError<br>        =====================<br>        <br>        Introduction<br>        ------------<br>        <br>        Flask-ReportableError is a Flask extension for handling errors that can<br>        be reported to the web client.<br>        <br>        <br>        Documentation<br>        -------------<br>        <br>        In order to use Flask-ReportableError, you must include the following<br>        statement in you application start script::<br>        <br>            from flask import Flask<br>            import flask_reportable_error<br>        <br>            app = Flask(__name__)<br>            app.config.from_envvar('FLASK_SETTINGS', silent=True)<br>            flask_reportable_error.init(app)<br>        <br>        <br>        API<br>        ---<br>        <br>        - ``flask_reportable_error.init()``:<br>          context initializer, it must receive the application as parameter.<br>        <br>        <br>        - ``flask_reportable_error.mixin``:<br>          class decorator that declares the decorated class as mixin with<br>          reportable exceptions.<br>        <br>        <br>        - ``flask_reportable_error.reportable()``:<br>          factory to create reportable exception classes. For example::<br>        <br>            raise reportable(ValueError)('invalid data received')<br>        <br>        <br>        - ``flask_reportable_error.ReportableErrorMixin``:<br>          mixin for reportable exception classes.<br>        <br>          - ``report()``:<br>            method that returns the reportable string \u2013\u00a0can be overridden.<br>        <br>          - ``status_code``:<br>            property representing the numeric status code \u2013\u00a0can be set at<br>            instance level.<br>        <br>          - ``template``:<br>            attribute refering to custom template name.<br>        <br>          - ``headers``:<br>            attribute refering to custom headers.<br>        <br>          - ``type_name``:<br>            attribute set by ``reportable()`` on inheritance, with the name of<br>            the original exception class.<br>        <br>        <br>        Settings<br>        --------<br>        <br>        The Flask settings may contain the key ``REPORTABLE_ERROR``, that\u2019s a<br>        dictionary with the following keys:<br>        <br>        - ``LOGLEVEL``:<br>          the logging level. If not supplied, Flask-ReportableError uses<br>          ``logging.ERROR``.<br>        <br>        - ``DEFAULT_STATUS_CODE``:<br>          the default numeric status code for reportable exception classes. By<br>          default it\u2019s 500.<br>        <br>        - ``TEMPLATE``:<br>          the name of a template to be used instead of none. The context of the<br>          template will contain the exception as value of the ``exc`` key.<br>        <br>        - ``HEADERS``:<br>          a dictionary to be used as default headers instead of ``{}``.\n          </div>"}, "last_serial": 989660, "releases": {"0.4.0": [{"comment_text": "", "digests": {"md5": "3bed69b64508cacb9d538e9be56a260e", "sha256": "16b4fff43e858dea89571532e2599231978f86ff4d1823add0da388e1496af8c"}, "downloads": -1, "filename": "Flask-ReportableError-0.4.0.tar.gz", "has_sig": false, "md5_digest": "3bed69b64508cacb9d538e9be56a260e", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 3056, "upload_time": "2013-10-17T19:11:35", "upload_time_iso_8601": "2013-10-17T19:11:35.584387Z", "url": "https://files.pythonhosted.org/packages/15/e0/7a406614c8768d164ccbb5e13cfeba6faa5d7027e3cad955e04ffc9af060/Flask-ReportableError-0.4.0.tar.gz", "yanked": false}], "0.4.1": [{"comment_text": "", "digests": {"md5": "77cb38a2541ffa8151813948cb071666", "sha256": "f75a45e890ae992c07347d9f2a955c1b24547b233bc97db1d9a36520d495eb2e"}, "downloads": -1, "filename": "Flask-ReportableError-0.4.1.tar.gz", "has_sig": false, "md5_digest": "77cb38a2541ffa8151813948cb071666", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 3059, "upload_time": "2013-10-18T15:11:04", "upload_time_iso_8601": "2013-10-18T15:11:04.939367Z", "url": "https://files.pythonhosted.org/packages/57/10/a11ded3bf0617caab71c83cfc23e8ecd0b48555bfd8fd69e3d620b10ede2/Flask-ReportableError-0.4.1.tar.gz", "yanked": false}], "0.4.2": [{"comment_text": "", "digests": {"md5": "dbe664b6fc851cad274577fc27241429", "sha256": "c8a8c9dea537159b317b36e320ad227a07715811f294e5583996b08dd5e8d724"}, "downloads": -1, "filename": "Flask-ReportableError-0.4.2.tar.gz", "has_sig": false, "md5_digest": "dbe664b6fc851cad274577fc27241429", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 3128, "upload_time": "2013-10-22T18:10:44", "upload_time_iso_8601": "2013-10-22T18:10:44.413007Z", "url": "https://files.pythonhosted.org/packages/6a/df/aa6f20f7cabc88db2dadfcf4949666530a22ac52f91bdd6b1f339bc9f581/Flask-ReportableError-0.4.2.tar.gz", "yanked": false}], "0.4.3": [{"comment_text": "", "digests": {"md5": "9a0cc646003c9f10ad7ff28914b306bb", "sha256": "8d9425cc0481ec067f683097c6b22c7e7c5b566ec993f5d610ac8690e31a02d2"}, "downloads": -1, "filename": "Flask-ReportableError-0.4.3.tar.gz", "has_sig": false, "md5_digest": "9a0cc646003c9f10ad7ff28914b306bb", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 3147, "upload_time": "2014-02-03T18:46:27", "upload_time_iso_8601": "2014-02-03T18:46:27.174542Z", "url": "https://files.pythonhosted.org/packages/e0/e3/63e01ae5288f1d599a463fc83a9d8753bffd65bb7d991c21220db741f9ce/Flask-ReportableError-0.4.3.tar.gz", "yanked": false}]}, "urls": [{"comment_text": "", "digests": {"md5": "9a0cc646003c9f10ad7ff28914b306bb", "sha256": "8d9425cc0481ec067f683097c6b22c7e7c5b566ec993f5d610ac8690e31a02d2"}, "downloads": -1, "filename": "Flask-ReportableError-0.4.3.tar.gz", "has_sig": false, "md5_digest": "9a0cc646003c9f10ad7ff28914b306bb", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 3147, "upload_time": "2014-02-03T18:46:27", "upload_time_iso_8601": "2014-02-03T18:46:27.174542Z", "url": "https://files.pythonhosted.org/packages/e0/e3/63e01ae5288f1d599a463fc83a9d8753bffd65bb7d991c21220db741f9ce/Flask-ReportableError-0.4.3.tar.gz", "yanked": false}], "timestamp": "Fri May  8 01:02:27 2020"}